Foundation/NSValue.swift

```diff
@@ -154,3 +154,14 @@ open class NSValue : NSObject, NSCopying, NSSecureCoding, NSCoding {
     }
 }
 
+extension NSValue : _Factory {}
+
+protocol _Factory {
+    init(factory: () -> Self)
+}
+
+extension _Factory {
+    init(factory: () -> Self) {
+        self = factory()
```

Comment:
ok, just prototyped a test for this... wow! this will solve a ton of issues with swift-corelibs-foundation!